hexl/ntt/fwd-ntt-avx512.cpp

```diff
@@ -183,6 +184,7 @@ void FwdT4(uint64_t* operand, __m512i v_neg_modulus, __m512i v_twice_mod,
   }
 }
 
+// In-place implementation
 template <int BitShift, bool InputLessThanMod>
 void FwdT8(uint64_t* operand, __m512i v_neg_modulus, __m512i v_twice_mod,
```

Comment:
If there's no performance penalty, could we have the in-place `FwdT8` call the out-of-place `FwdT8`?

Reply:
Benchmarking results have been very irregular. I discarded many runs because results looked random. Anyway, here are the new results, I marked with an asterisk the ones that seems to have an impact. Let me know what you think about the results.
[Performance.xlsx](https://github.com/intel/hexl/files/7281466/Performance.xlsx)